### DISCUSSION TITLE
Confirming NativeCamera video fails on first try

### DISCUSSION THREAD

**Discussion by romanvanderaa** [yasirkula/UnityNativeCamera#111] [Q&A]
I use NativeCamera in my application to record a video, and save it to a specific folder inside the app's storage. 

I'm using a selfmade file manager class, which doesn't have any problems saving images or anything, but when I try to record and save a video using NativeCamera it fails the first time. 

The first video I make, and tap OK to confirm, doesn't have a file path. Is anyone experiencing the same issue?

I'll add the code to record and save the video, and the part from my file manager that saves the video:

Video recording:
`public void StartVideoCapture()
    {
        NativeCamera.Permission permission = NativeCamera.RecordVideo((path) =>
        {
            Debug.Log("Path: " + path);
            if (path != null)
            {
                SaveVideo(path);
            }
            else
            {
                Debug.LogError("Returned null path");
            }
        }, NativeCamera.Quality.High, 500);  
    }

    void SaveVideo(string videoPath)
    {
        if (string.IsNullOrEmpty(videoPath))
        {
            Debug.LogError("Video path is empty");
            return;
        }

        FileManager.SaveWideVideo(DataProvider.selectedAsset, DataProvider.selectedAcquisition, videoPath);
        statusText.text = "Video saved successfully";
    }`

File management:
`public static void SaveWideVideo(Data.AssetObject asset, Data.AcquisitionObject acquisition, string sourcePath)
    {
        CheckAllDirectories(asset, acquisition);

        string folderPath = GetWideImagesDir(asset, acquisition); 
        string videoDestPath = Path.Combine(folderPath, "video_wide.mp4"); 

        File.Copy(sourcePath, videoDestPath, true);
        Debug.Log("Video saved at: " + videoDestPath);
    }`

**Comment by yasirkula** [comment]
Is the path returned by NativeCamera itself null or does it get null somewhere inside SaveWideVideo? And can you spot any related native logs?
